test/integration_test.rb

```diff
@@ -0,0 +1,62 @@
+require 'helpers/kubeclient_helper'
+
+module KubernetesDeploy
+  class IntegrationTest < ::Minitest::Test
+    include KubeclientHelper
+
+    def setup
+      @logger_stream = StringIO.new
+      @logger = Logger.new(@logger_stream)
+      KubernetesDeploy.logger = @logger
+      @namespace = TestProvisioner.claim_namespace
+    end
+
+    def teardown
+      @logger_stream.close
+      TestProvisioner.delete_namespace(@namespace)
+    end
+  end
+
+  module TestProvisioner
+    extend KubeclientHelper
+
+    def self.claim_namespace
+      ns = SecureRandom.hex(8)
+      create_namespace(ns)
+      ns
+    rescue KubeException => e
+      retry if e.to_s.include?("already exists")
+    end
+
+    def self.create_namespace(namespace)
+      ns = Kubeclient::Namespace.new
+      ns.metadata = { name: namespace }
+      kubeclient.create_namespace(ns)
+    end
+
+    def self.delete_namespace(namespace)
+      kubeclient.delete_namespace(namespace)
+    rescue KubeException => e
+      raise unless e.to_s.include?("not found")
+    end
+
+    def self.prepare_pv(name)
+      begin
+        kubeclient.get_persistent_volume(name)
+      rescue KubeException => e
+        raise unless e.to_s.include?("not found")
+        pv = Kubeclient::PersistentVolume.new
+        pv.metadata = { name: name }
+        pv.spec = {
+          accessModes: ["ReadWriteOnce"],
+          capacity: { storage: "1Gi" },
+          hostPath: { path: "/data/#{name}" },
+          persistentVolumeReclaimPolicy: "Recycle"
+        }
+        kubeclient.create_persistent_volume(pv)
+      end
+    end
+  end
+
+  TestProvisioner.prepare_pv("pv0001")
```

Comment:
Tests won't reference this by name. It's a cluster-admin-level resource, which is why we aren't even cleaning it up within or after runs. Here's a useful snippet on how these work:

> A user creates, ..., a PersistentVolumeClaim with a specific amount of storage requested and with certain access modes. A control loop in the master watches for new PVCs, finds a matching PV (if possible), and binds them together. ..., the user will always get at least what they asked for, but the volume may be in excess of what was requested. Once bound, PersistentVolumeClaim binds are exclusive, regardless of the mode used to bind them.

So the tests / yamls need to match the `accessMode` of an existing PV and request <= the amount of storage it has, but they don't need to know its name.